### PR TITLE
remove default tls keys

### PR DESCRIPTION
**Pull request by chaospuppy**
...and expand options for supplying/autogenerating them.

Having the [public token signing keys](https://github.com/goharbor/harbor-helm/tree/master/cert) used by default essentially makes the Harbor Helm chart insecure by default.  It is easy to miss that if you do not specify a preexisting `core.secretName` value, that a publicly disclosed key will be used to sign tokens issued from the Harbor API.

This PR will remove those certificates, adds a new `core.tokenKey` value where a PEM-encoded RSA key can be specified, and if neither `core.tokenKey` or `core.secretName` are provided, the helm template for the core secret will autogenerate a unique RSA key.

Additionally, this PR removes the `tls.crt` key from secrets, as it is not used anywhere, and also removes a portion of the `harbor-exporter` secret, which also was not used. 

Signed-off-by: Tim Seagren <seagren.tim@solute.us>

### CODE REVIEW

**Review by stonezdj — APPROVED** [review]
LGTM

File: values.yaml [hunk 1]
```diff
@@ -498,10 +498,10 @@ core:
   # Fill the name of a kubernetes secret if you want to use your own
   # TLS certificate and private key for token encryption/decryption.
   # The secret must contain keys named:
-  # "tls.crt" - the certificate
   # "tls.key" - the private key
-  # The default key pair will be used if it isn't set
   secretName: ""
+  # If not specifying a preexisting secret, a secret can be created from tokenKey and tokenCertificate and used instead. If neither secretName or tokenKey are specified, a certificate will be autogenerated.
```
Comment (zyyw) [inline values.yaml:503]: Please add a comment to emphasize the following two points, if using `tokenKey`:
* the content should be a multi-line string with one more indentation of `tokenKey`
* the content of `tokenKey` should be a `rsa` type privateKey

Comment (zyyw) [reply values.yaml:503]: @chaospuppy thank you for contributing to harbor-helm. Please help to update this PR to resolve the above comment.
Appreciate it!

**Comment by zyyw** [comment]
When I was trying to double check this PR (git checkout to the [chaospuppy:remove-default-keypair](https://github.com/chaospuppy/harbor-helm/tree/remove-default-keypair) branch) by installing harbor-helm with the following command:
```
helm install harbor-v263 --set persistence.enabled=false,externalURL=https://<harbor-ip>:30003,expose.type=nodePort,expose.tls.auto.commonName=<harbor-ip>,imagePullSecrets[0].name=dockerhub-cred .
```
Notary pod CrashLoopBackOff:
```
ubuntu@shengwen:~/k8s-kind/harbor-helm$ kubectl get pods
NAME                                                       READY   STATUS             RESTARTS      AGE
harbor-no-default-keypair-chartmuseum-79d6999d88-lb6gn     1/1     Running            0             11m
harbor-no-default-keypair-core-6689bb96f9-qghzf            1/1     Running            0             11m
harbor-no-default-keypair-database-0                       1/1     Running            0             11m
harbor-no-default-keypair-jobservice-577598f7cf-plw7d      1/1     Running            2 (11m ago)   11m
harbor-no-default-keypair-nginx-7dc8bffbc9-7s266           1/1     Running            0             11m
harbor-no-default-keypair-notary-server-6c794d8694-hzkt6   0/1     CrashLoopBackOff   7 (28s ago)   11m
harbor-no-default-keypair-notary-signer-d58975fc5-pf7j5    1/1     Running            0             11m
harbor-no-default-keypair-portal-7c57c4f6ff-nkq2n          1/1     Running            0             11m
harbor-no-default-keypair-redis-0                          1/1     Running            0             11m
harbor-no-default-keypair-registry-756b4b4595-vmcf8        2/2     Running            0             11m
harbor-no-default-keypair-trivy-0                          1/1     Running            0             11m
```
And describe the pod, it says 
```
ubuntu@shengwen:~/k8s-kind/harbor-helm$ kubectl describe pod harbor-no-default-keypair-notary-server-6c794d8694-hzkt6
Name:         harbor-no-default-keypair-notary-server-6c794d8694-hzkt6
Namespace:    default
Priority:     0
Node:         kind-control-plane/172.18.0.2
Start Time:   Thu, 16 Feb 2023 11:44:18 +0000
Labels:       app=harbor
              chart=harbor
              component=notary-server
              heritage=Helm
              pod-template-hash=6c794d8694
              release=harbor-no-default-keypair
Annotations:  checksum/secret: eac0d713800e0251937e7f422af9f575101538aa611af7b4d7831b579e810573
              checksum/secret-core: e22b94cb06e342ba8549fe88434ffacf864932a44fb701e911f8c984ddffb595
Status:       Running
IP:           10.244.0.116
IPs:
  IP:           10.244.0.116
Controlled By:  ReplicaSet/harbor-no-default-keypair-notary-server-6c794d8694
Containers:
  notary-server:
    Container ID:   containerd://40671aff7b3ea5e77e59f7086674871488cceb31dbcea724fa7e0095ef5e82b6
    Image:          goharbor/notary-server-photon:dev
    Image ID:       docker.io/goharbor/notary-server-photon@sha256:0b19c0347e2e2ca823add3ad2c3f27a0bbb055c25c4561a8dcdad67424dc960a
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Thu, 16 Feb 2023 11:47:21 +0000
      Finished:     Thu, 16 Feb 2023 11:47:22 +0000
    Ready:          False
    Restart Count:  5
    Liveness:       http-get http://:4443/_notary_server/health delay=300s timeout=1s period=10s #success=1 #failure=3
    Readiness:      http-get http://:4443/_notary_server/health delay=20s timeout=1s period=10s #success=1 #failure=3
    Environment:
      MIGRATIONS_PATH:  migrations/server/postgresql
      DB_URL:           <set to the key 'NOTARY_SERVER_DB_URL' in secret 'harbor-no-default-keypair-notary-server'>  Optional: false
    Mounts:
      /etc/notary/server-config.postgres.json from config (rw,path="server.json")
      /etc/ssl/notary/ca.crt from signer-certificate (rw,path="ca.crt")
      /root.crt from token-service-certificate (rw,path="tls.crt")
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-no-default-keypair-notary-server
    Optional:    false
  token-service-certificate:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-no-default-keypair-core
    Optional:    false
  signer-certificate:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-no-default-keypair-notary-server
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason       Age                    From               Message
  ----     ------       ----                   ----               -------
  Normal   Scheduled    5m42s                  default-scheduler  Successfully assigned default/harbor-no-default-keypair-notary-server-6c794d8694-hzkt6 to kind-control-plane
  Warning  FailedMount  5m41s                  kubelet            MountVolume.SetUp failed for volume "config" : failed to sync secret cache: timed out waiting for the condition
  Warning  FailedMount  5m41s                  kubelet            MountVolume.SetUp failed for volume "signer-certificate" : failed to sync secret cache: timed out waiting for the condition
  Normal   Pulled       4m46s (x4 over 5m39s)  kubelet            Container image "goharbor/notary-server-photon:dev" already present on machine
  Normal   Created      4m46s (x4 over 5m39s)  kubelet            Created container notary-server
  Normal   Started      4m46s (x4 over 5m39s)  kubelet            Started container notary-server
  Warning  BackOff      33s (x32 over 5m31s)   kubelet            Back-off restarting failed container
ubuntu@shengwen:~/k8s-kind/harbor-helm$
```

Deployed harbor-helm from master branch, it succeeded. 
Rebase master into remove-default-keypair branch and then deploy harbor via harbor-helm, it still reports same error which is:
```
harbor-rebased-no-keypair-notary-server-cc7649fb5-mhs96   0/1     CrashLoopBackOff   4 (58s ago)     2m28s
```
kubectl describe the notary-server pod, same error message:
```
Events:
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    2m40s                 default-scheduler  Successfully assigned default/harbor-rebased-no-keypair-notary-server-cc7649fb5-mhs96 to kind-control-plane
  Warning  FailedMount  2m39s                 kubelet            MountVolume.SetUp failed for volume "signer-certificate" : failed to sync secret cache: timed out waiting for the condition
  Warning  FailedMount  2m39s                 kubelet            MountVolume.SetUp failed for volume "config" : failed to sync secret cache: timed out waiting for the condition
  Normal   Pulled       115s (x4 over 2m38s)  kubelet            Container image "goharbor/notary-server-photon:dev" already present on machine
  Normal   Created      115s (x4 over 2m38s)  kubelet            Created container notary-server
  Normal   Started      115s (x4 over 2m38s)  kubelet            Started container notary-server
  Warning  BackOff      85s (x11 over 2m28s)  kubelet            Back-off restarting failed container
```

File: values.yaml (hunk 1 above)
Comment (chaospuppy) [reply values.yaml]: Updated comments and addressed @zyyw's error

**Comment by chaospuppy** [comment]
Thank you for pointing that out @zyyw, I had missed that the `tls.crt` key of the harbor core secret was being used by Notary to validate token signatures.  Updated the core secret to generate a CA, with the CA key used as `tls.key` and the CA cert used as `tls.crt` if none of `secretName`, `tokenCert` and `tokenKey` are supplied.

**Comment by zyyw** [comment]
Hi @chaospuppy could you please also help to:
* add README for the two new configurations introduced in values.yaml. I mean add README for `tokenKey` and `tokenCert`.
* squash your commits into one commit

Thanks for the help. Really appreciate it!

**Comment by chaospuppy** [comment]
@zyyw added docs, rebased from master, and squashed all commits.  No problem!

**Review by Vad1mo — APPROVED** [review]

**Comment by OrlinVasilev** [comment]
@chaospuppy - congrats on your first merged PR :))

**Pull request merged**